config/config.go

```diff
@@ -668,6 +668,12 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	if len(c.Masters) == 0 {
 		return fmt.Errorf("Kubernetes SD configuration requires at least one Kubernetes master")
 	}
+	if len(c.BearerToken) > 0 && len(c.BearerTokenFile) > 0 {
+		return fmt.Errorf("at most one of bearer_token & bearer_token_file must be configured")
```

Comment:
That doesn't make this check incorrect though, right? you're saying there should be an additional validation on this section?

protocol wise, cert & bearer token auth are not necessary mutually exclusive, but you're right kubernetes implements them as separate auth options.